content/40.units_and_quantities.md

```diff
@@ -1,269 +1,682 @@
+dimensions of `Msun / Mpc**3` are `mass / length**3` and the CGS conversion factor is about $6.8
+\times 10^{-41}$.
+
+We use the `UnitRegistry` class to define all valid atomic units. All unit registries contain a unit
+symbol lookup table (in a Python `OrderedDict`) containing for each unit in a tuple the conversion factor to SI units, the dimensionality, the additive offset for the unit, a string representing how the unit should be displayed in plots and other media, and a boolean dictating whether the unit is SI-prefixable. 
+Here is what it would look like with the above units:
+
+```python
+{ 
+    "g":    (1.0e-3, dimensions.mass, 0.0, r"\rm{g}", True),
+    "erg":  (1.0e-7, dimensions.energy, 0.0, r"\rm{erg}", True),
+    "Msun": (mass_sun_kg, dimensions.mass, 0.0, r"\rm{M}_\odot", False),
+    "pc":   (m_per_pc, dimensions.length, 0.0, r"\rm{pc}", True), 
+    "degC": (1.0, dimensions.temperature, -273.15, r"^\circ\rm{C}", True),
+}
+```
+
+Note that we only define **atomic** units here. There should be no operations or SI prefixes in the
+registry symbol strings. When we parse non-atomic units like `Msun/Mpc**3`, we use the registry to
+look up the symbols. The unit system in `yt` knows how to handle units like `Mpc` by looking up unit
+symbols with and without prefixes and modify the conversion factor appropriately. Note also that in most cases the "additive offset" will be 0--it applies mainly to temperature scales like Fahrenheit and Celsius where the zero-point of the scale are not equivalent between the two scales, and are also not equivalent to "zero thermal energy", for which the relevant temperature units are Rankine and Kelvin, respectively.
+
+We construct a `Unit` object by providing a string containing atomic unit symbols, combined with operations in Python syntax, and the registry those atomic unit symbols are defined in. We use
+`sympy`'s string parsing features to create the unit expression from the user-provided string. Each `Unit`
+object also has its own unit `sympy` `Expr` object, a dimensionality `Expr` object, a `UnitRegistry` instance, a scalar conversion factor to SI units, additive offset value, and display representation. Among other methods, `Unit` defines the `mul`, `div`, `pow`, and `eq` operations with
+other unit objects, making it easy to compose compound units algebraically.
+
+The `UnitRegistry` class provides the `add`, `remove`, and `modify` methods which allows users to
+add, remove, and modify atomic unit definitions present in `UnitRegistry` objects. In general, unit registries should only be adjusted inside of a code frontend, since otherwise quantities and arrays might be created with inconsistent unit metadata. Once a `Unit` object is created, it will not receive updates if the original unit registry is modified.
+
+#### Code units {#sec:code-units}
+
+When a `Dataset` object is instantiated, it will itself instantiate and set up a `UnitRegistry`
```

Comment:
(cultural context: repetition is the bane of French teachers. This is probably why I'm chasing them, but I can imagine it's not that big of a deal in English)
```suggestion
When a `Dataset` object is created, it will itself instantiate and set up a `UnitRegistry`
```